src/Groups/matrices/MatGrp.jl

```diff
@@ -505,7 +505,15 @@ function gens(G::MatrixGroup)
    return G.gens
 end
 
-gen(G::MatrixGroup, i::Int) = gens(G)[i]
+# Note that the `gen(G::GAPGroup, i::Int)` method cannot be used
+# for `MatrixGroup` because of the `:gens` attribute.
+function gen(G::MatrixGroup, i::Int)
+  i == 0 && return one(G)
+  L = gens(G)
+  0 < i && i <= length(L) && return L[i]
+  i < 0 && -i <= length(L) && return inv(L[-i])
```

Comment:
I guess on the long run the idea would be to cache this inverse, esp. for matrix groups, where it can be expensive to compute. But that can come in a future PR.